hydrator-common/src/main/java/io/cdap/plugin/common/db/AbstractDBConnector.java

```diff
@@ -106,6 +106,10 @@ public void test(ConnectorContext context) throws ValidationException {
 
   @Override
   public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest request) throws IOException {
+    return this.browse(request);
```

Comment:
I think it might not be correct and we should be using `ConnectorContext` and store the errors in `FailureCollector` as done in `test(ConnectorContext context)`: https://github.com/cdapio/hydrator-plugins/blob/339807e901b82ad1605d3fbc1c90bec76f14481f/hydrator-common/src/main/java/io/cdap/plugin/common/db/AbstractDBConnector.java#L102

@albertshau and @yaojiefeng wdyt??

Reply:
FailureCollector is used primarily when validating, to highlight multiple errors at the same time, or to include more information in the response like the stack trace. 

In this particular case, it doesn't look like it matters much since there would only be one generic error. It's probably still better to use the FailureCollector since we can add the stack trace to the failure and it doesn't really complicate the logic much.

Reply:
As Rahul pointed out offline, the ConnectionHandler in CDAP doesn't actually do anything with the failure collector, so we would have to fix that before actually using it anywhere